packages/transactions/src/keys.ts

```diff
@@ -139,45 +159,51 @@ export interface StacksPrivateKey {
 
 export function createStacksPrivateKey(key: string | Buffer): StacksPrivateKey {
   const data = typeof key === 'string' ? Buffer.from(key, 'hex') : key;
-  let compressed: boolean;
-  if (data.length === 33) {
-    if (data[data.length - 1] !== 1) {
-      throw new Error(
-        'Improperly formatted private-key. 33 byte length usually ' +
-          'indicates compressed key, but last byte must be == 0x01'
-      );
-    }
-    compressed = true;
-  } else if (data.length === 32) {
-    compressed = false;
-  } else {
-    throw new Error(
-      `Improperly formatted private-key hex string: length should be 32 or 33 bytes, provided with length ${data.length}`
-    );
-  }
+  const compressed: boolean = isPrivateKeyCompressed(key);
   return { data, compressed };
 }
 
 export function makeRandomPrivKey(entropy?: Buffer): StacksPrivateKey {
-  const ec = new EC('secp256k1');
-  const options = { entropy: entropy || randomBytes(32) };
-  const keyPair = ec.genKeyPair(options);
-  const privateKey = keyPair.getPrivate().toString('hex', 32);
-  return createStacksPrivateKey(privateKey);
+  // Extended form of secp.utils.randomPrivateKey() to support entropy
+  const bytes: Uint8Array = new Uint8Array(entropy?.length || 0);
+  if (entropy) {
+    for (let i = 0; i < entropy.length; i++) bytes[i] = entropy[i];
+  }
+  let i = 8;
+  while (i--) {
+    const b32 = bytes.length ? bytes : utils.randomBytes(32);
+    const num = BigInt(`0x${utils.bytesToHex(b32)}`);
+    // Check isWithinCurveOrder
+    if (BigInt(0) < num && num < CURVE.n && num !== BigInt(1))
+      return createStacksPrivateKey(utils.bytesToHex(b32));
+  }
+  throw new Error('Valid private key was not found in 8 iterations. PRNG is broken');
+}
+
+function verifyHex(input: string): string {
+  const regexp = /^[0-9a-fA-F]+$/;
+  let hex = input;
+  if (!regexp.test(input)) {
+    hex = utils.bytesToHex(utf8ToBytes(input));
```

Comment:
The name of the function is inaccurate, and the functionality is not clear to me. It appears to be a heuristic to detect hex strings, and if a non-hex is detected then it treats the input as utf8? That is surprising behavior which should be avoided.

Reply:
For example, if this function is being used for the utf8 functionality, then a message like `1234` will be detected as hex.